tests/test_queries.py

```diff
@@ -0,0 +1,53 @@
+import pytest
+import torch
+from perceiver_pytorch.queries import LearnableQuery
+from perceiver_pytorch.perceiver_io import PerceiverIO
+import einops
+
+
+@pytest.mark.parametrize("layer_shape", ["2d", "3d"])
+def test_learnable_query(layer_shape):
+    query_creator = LearnableQuery(
+        channel_dim=32,
+        query_shape=(6, 16, 16),
+        conv_layer=layer_shape,
+        max_frequency=64.0,
+        frequency_base=2.0,
+        num_frequency_bands=128,
+        sine_only=False,
+    )
+    x = torch.randn((4, 6, 12, 16, 16))
+    out = query_creator(x)
+    assert out.shape == (4, 1536, 803)
```

Comment:
Are these dimensions correct? Can you add a comment showing why these are right?

Reply:
Yes, and yeah, just did